homeassistant/components/songpal/config_flow.py

```diff
@@ -147,7 +147,7 @@ async def async_step_import(self, user_input=None):
 
     def _endpoint_already_configured(self):
         """See if we already have an endpoint matching user input configured."""
-        existing_endpoints = [
-            entry.data[CONF_ENDPOINT] for entry in self._async_current_entries()
-        ]
-        return self.conf.endpoint in existing_endpoints
+        for entry in self._async_current_entries():
+            if entry.data.get(CONF_ENDPOINT) == self.conf.endpoint:
```

Comment:
If `_async_current_entries` is a coroutine as the name indicates, shouldn't this use `async for` here? Otherwise LGTM :+1:

Reply:
`_async_current_entries` is marked as a callback, which is a sync function safe to call from whithin the event loop. Maybe `_endpoint_already_configured` should be marked as a callback as well.